internal/command/ssh/issue.go

```diff
@@ -97,24 +106,22 @@ func runSSHIssue(ctx context.Context) (err error) {
 		emails   string
 		rootname string
 	)
-	switch args := flag.Args(ctx); len(args) {
-	case 0:
-		// neither
-	case 1:
-		// org only
-	case 2:
+
+	orgFlag := flag.GetOrg(ctx)
+	email := flag.GetString(ctx, "email")
+	path := flag.GetString(ctx, "path")
+
+	if orgFlag != "" && path != "" && email != "" {
+		// org+email+path
+		emails = email
+		rootname = path
+	} else if orgFlag != "" || email != "" {
 		// org+email or org+path
-		if _, err = mail.ParseAddress(args[1]); err == nil {
-			emails = args[1]
+		if _, err = mail.ParseAddress(email); err == nil {
+			emails = email
 		} else {
-			rootname = args[1]
+			rootname = orgFlag
 		}
-	case 3:
-		// org+email+path
-		emails = args[1]
-		rootname = args[2]
-	default:
-		return errors.New("Too many positional arguments\n")
 	}
```

Comment:
I am afraid I don't have time at the moment but this whole thing looks shady to me. Needs 🧠 power and settle down on how this positional values that now are flags were used to convey the different command combinations.

Reply:
I think the argument changes started with https://github.com/superfly/flyctl/pull/1572.
I can do a more in-depth analysis of this tomorrow.

Reply:
First, the issue command supported `[org] [email] [path]` args.
And `--org`, `--username`, `--hours`, `--agent`, `--dotssh`, `--overwrite` flags.

PR #1572 made the following changes related to the args of `fly issue ssh` command:
- `--username` flag to accept a string slice instead of string.

- Adds comment saying API args username and email are deprecated. Removes `[email]` arg,
but still uses its value in the switch statement?

- Gets just the name part of the email address and adds it to the principals.

I think the only issue in my code is changing the addition in line 123 to be `email`, not `orgFlag`
as `[email]` arg was `args[1]` previously.

Reply:
Actually, after removing the `[email]` arg,
the code seems to expect the value of the second arg to be used for `[email]` or `[path]`.